### PR TITLE
DTSCCI-903 Consolidate work arounds that re-establish Case Flags on case data for LR events

### PR DESCRIPTION
**Pull request by tiagreaney**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DTSCCI-903

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/AcknowledgeClaimCallbackHandlerTest.java [hunk 1]
```diff
@@ -323,17 +354,25 @@ void shouldCopyRespondentPrimaryAddresses_whenInvoked() {
...
+            CaseData oldCaseData = CaseDataBuilder.builder()
+                .applicant1(Party.builder().partyName("name").type(INDIVIDUAL).primaryAddress(testAddress).build())
+                .respondent1(Party.builder().partyName("name").type(INDIVIDUAL).primaryAddress(testAddress).build())
+                .build();
+            when(caseDetailsConverter.toCaseData(any(CaseDetails.class))).thenReturn(oldCaseData);
 
             var response = (AboutToStartOrSubmitCallbackResponse) handler.handle(params);
 
-            assertThat(response.getData()).doesNotContainKey("respondent1Copy");
+            //assertThat(response.getData()).doesNotContainKey("respondent1Copy");
```
Comment (ruban72) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/AcknowledgeClaimCallbackHandlerTest.java:375]: Can we remove this commented out code if not required?

**Review by ruban72 — CHANGES_REQUESTED** [review]
Looks like there is compile issue:
https://github.com/hmcts/civil-service/actions/runs/11443682812/job/31836878123?pr=5662

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/AcknowledgeClaimCallbackHandler.java [hunk 2]
```diff
@@ -121,10 +116,11 @@ private CallbackResponse populateRespondentCopyObjects(CallbackParams callbackPa
         if (solicitorRepresentsOnlyOneOrBothRespondents(callbackParams, RESPONDENTSOLICITORTWO)) {
             isRespondent1 = NO;
         }
-        updatedCaseData.solicitorReferencesCopy(caseData.getSolicitorReferences());
-        updatedCaseData.isRespondent1(isRespondent1);
+        var caseDateBuilder = caseData.toBuilder();
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/AcknowledgeClaimCallbackHandler.java:119]: Can you not do it as a one liner? caseData.toBuilder().solicitorReferencesCopy(etc).etc().build()

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/AcknowledgeClaimCallbackHandler.java [hunk 3]
```diff
@@ -276,17 +262,14 @@ && respondent2HasSameLegalRep(caseData)) {
 
         } else if (caseData.getAddRespondent2() != null && caseData.getAddRespondent2().equals(YES)
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/AcknowledgeClaimCallbackHandler.java:263]: Could use the opportunity to refactor a bit like this conditional could be extracted to a method for better readability

**Review by gmikerlemos — DISMISSED** [review]

**Review by gmikerlemos — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimSpecCallbackHandler.java [hunk 4]
```diff
@@ -1430,22 +1423,16 @@ && ifResponseTypeIsPartOrFullAdmission(caseData)) {
...
                 Party updatedRespondent2;
 
                 if (NO.equals(caseData.getSpecAoSRespondent2HomeAddressRequired())) {
                     updatedRespondent2 = caseData.getRespondent2().toBuilder()
                         .primaryAddress(caseData.getSpecAoSRespondent2HomeAddressDetails()).build();
                 } else {
-                    updatedRespondent2 = caseData.getRespondent2().toBuilder()
-                        .primaryAddress(caseData.getRespondent2Copy().getPrimaryAddress()).build();
+                    CaseData.CaseDataBuilder caseDataBuilder = caseData.toBuilder();
+                    PersistDataUtils.persistFlagsForParties(caseData, oldCaseData, caseDataBuilder);
```
Comment (velimd) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimSpecCallbackHandler.java:1434]: shouldnt this be persistPartyAddress()?

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimSpecCallbackHandler.java [hunk 5]
```diff
@@ -1430,22 +1423,16 @@ && ifResponseTypeIsPartOrFullAdmission(caseData)) {
...
                 }
-
-                updatedData
-                    .respondent2(updatedRespondent2.toBuilder()
-                                     .flags(caseData.getRespondent2Copy().getFlags()).build())
-                    .respondent2Copy(null);
-                updatedData.respondent2DetailsForClaimDetailsTab(updatedRespondent2.toBuilder().flags(null).build());
```
Comment (velimd) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimSpecCallbackHandler.java:1448]: I feel like this persistFlagsForParties method should be here no due to the case flag issue?

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimCallbackHandler.java [hunk 6]
```diff
@@ -475,16 +465,8 @@ && ofNullable(caseData.getRespondent2Copy()).isPresent()) {
...
-                var updatedRespondent2 = caseData.getRespondent2().toBuilder()
-                    .primaryAddress(caseData.getRespondent2Copy().getPrimaryAddress())
-                    .build();
-
-                updatedCaseDataBuilder
-                    .respondent2(updatedRespondent2)
-                    .respondent2Copy(null)
-                    .respondent2DetailsForClaimDetailsTab(updatedRespondent2.toBuilder().flags(null).build());
+            if (ofNullable(caseData.getRespondent2()).isPresent()) {
+                PersistDataUtils.persistPartyAddress(oldCaseData, caseData);
```
Comment (velimd) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimCallbackHandler.java:469]: What about the respondent2DetailsForClaimDetailsTab? this part doesnt consider it right?

**Pull request closed without merging**